scala/wdlscala/src/main/scala/wdlscala/Workflow.scala

```diff
@@ -0,0 +1,174 @@
+      case (_, terminals) if terminals.size > 1 =>
+        throw new SyntaxError(wdlSyntaxErrorFormatter.multipleCallsAndHaveSameName(terminals.asInstanceOf[Seq[Terminal]]))
+      case _ =>
+    }
+
+    new Workflow(name, declarations, workflowOutputsDecls)
+  }
+}
+
+/**
+ * Represents a `workflow` definition in WDL which currently
+ * only supports a set of `call` declarations and a name for
+ * the workflow
+ *
+ * @param unqualifiedName The name of the workflow
+ */
+case class Workflow(unqualifiedName: String,
+                    declarations: Seq[Declaration],
+                    workflowOutputDecls: Seq[WorkflowOutputDeclaration]) extends Executable with Scope {
+  override val prerequisiteScopes = Set.empty[Scope]
+  override val prerequisiteCallNames = Set.empty[String]
+  override val parent: Option[Scope] = None
+
+  /**
+   * FQNs for all inputs to this workflow and their associated types and possible postfix quantifiers.
+   *
+   * @return a Map[FullyQualifiedName, WorkflowInput] representing the
+   *         inputs that the user needs to provide to this workflow
+   */
+  def inputs: Map[FullyQualifiedName, WorkflowInput] = {
+    val callInputs = for { call <- calls; input <- call.unsatisfiedInputs } yield input
+    val declarationInputs = for { declaration <- declarations; input <- declaration.asWorkflowInput } yield input
+    (callInputs ++ declarationInputs) map { input => input.fqn -> input } toMap
+  }
+
+  /**
+   * All outputs for this workflow and their associated types
+   *
+   * @return a Map[FullyQualifiedName, WdlType] representing the union
+   *         of all outputs from all `call`s within this workflow
+   */
+  lazy val outputs: Seq[ReportableSymbol] = {
+
+    case class PotentialReportableSymbol(name: String, wdlType: WdlType, matchWorkflowOutputWildcards: Boolean)
+
+    // Build a list of ALL potentially reportable symbols, and whether they're allowed to match
+    // wildcards in the workflow's output {...} spec.
+    val outputs: Seq[PotentialReportableSymbol] = for {
+      call: Call <- calls
+      output <- call.task.outputs
+    } yield PotentialReportableSymbol(s"${call.fullyQualifiedName}.${output.name}", output.wdlType, matchWorkflowOutputWildcards = true)
+
+    val inputs: Seq[PotentialReportableSymbol] = for {
+      call: Call <- calls
+      input <- call.task.inputs
+    } yield PotentialReportableSymbol(s"${call.fullyQualifiedName}.${input.name}", input.wdlType, matchWorkflowOutputWildcards = false)
+
+    val filtered = if (workflowOutputDecls isEmpty) {
+      outputs
+    } else {
+      (outputs ++ inputs) filter {
+        case PotentialReportableSymbol(fqn, wdlType, wildcardsAllowed) =>
+          workflowOutputDecls.isEmpty || workflowOutputDecls.exists(_.outputMatchesDeclaration(fqn, wildcardsAllowed))
+      }
+    }
+
+    val x = filtered map { case PotentialReportableSymbol(fqn, value, wildcardAllowed) => ReportableSymbol(fqn, value) }
+    x
```

Comment:
This might cause me to overrule my ban on fixing things I didn't change. I curse you for bringing this to my attention